skorch/helper.py

```diff
@@ -256,3 +261,250 @@ def predefined_split(dataset):
 
     """
     return partial(_make_split, valid_ds=dataset)
+
+
+class DataFrameTransformer(BaseEstimator, TransformerMixin):
+    """Transform a DataFrame into a dict useful for working with skorch.
+
+    Transforms cardinal data to floats and categorical data to vectors
+    of ints so that they can be embedded.
+
+    Although skorch can deal with pandas DataFrames, the default
+    behavior is often not very useful. Use this transformer to
+    transform the DataFrame into a dict with all float columns
+    concatenated using the key "X" and all categorical values encoded
+    as integers, using their respective column names as keys.
+
+    Your module must have a matching signature for this to work. It
+    must accept an argument ``X`` for all cardinal
+    values. Additionally, for all categorical values, it must accept
+    an argument with the same name as the corresponding column (see
+    example below). If you need help with the required signature, use
+    the ``describe_signature`` method of this class and pass it your
+    data.
+
+    You can choose whether you want to treat int columns the same as
+    float columns (default) or as categorical values.
+
+    To one-hot encode categorical features, initialize their
+    corresponding embedding layers using the identity matrix.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({
+    ...     'col_floats': np.linspace(0, 1, 12),
+    ...     'col_ints': [11, 11, 10] * 4,
+    ...     'col_cats': ['a', 'b', 'a'] * 4,
+    ... })
+    >>> # cast to category dtype to later learn embeddings
+    >>> df['col_cats'] = df['col_cats'].astype('category')
+    >>> y = np.asarray([0, 1, 0] * 4)
+
+    >>> class MyModule(nn.Module):
+    ...     def __init__(self):
+    ...         super().__init__()
+    ...         self.reset_params()
+
+    >>>     def reset_params(self):
```

Comment:
Should this be directly in `__init__`?

Reply:
It's just my personal preference, that way it's possible to reset the module without creating a new instance.